### PR TITLE
Parameterized volume size support for Oracle classic builder

### LINKED ISSUE #6509: Increase the disk size used during a Packer builder for Oracle Compute (Classic)
Opening issue on behalf of a customer:

> We are using Packer to build an Oracle Compute Machine Image but one of the provisioners fails due to hitting out-of-disk. For most other builders it is very easy to increase disk but for oracle-classic builder, i can't see any such option.

### LINKED ISSUE #6833: Oracle Classic - Missing Source Image List Entry Value
### Provider
- oracle classic

### Feature request
The ability to set the image list entry for the source image list. This will allow us to select a specific version of an Oracle Provided image as the source.

### I tried to do it

I have tried to add this functionality myself, as the underlying  go-oracle-terraform/compute/instances.go does expose an "Entry" attribute, which is exactly what we want.

I added the following to config.go (around line 34)

```
SourceImageListEntry  string    `mapstructure:"source_image_list_entry"`
```

and added the following to set_create_instance.go (around line 36)

```
		Entry:      config.SourceImageListEntry,
```

However when I run packer I get

```
Failed to mapstructure Config: 1 error(s) occurred:

* unknown configuration key: "source_image_list_entry"
```

### PR DESCRIPTION
**Pull request by mwhooker**
Support arbitrary size volume for performing oracle classic build

I still need to do some clean up (read: rebasing) and decide if some of the names I chose are good, but I believe reviewing can start. Any feedback around names would be useful. @scross01 if you could also review, I would appreciate it.

Some things that still need to be done

- [x] documentation
- [x] support for linux builder and windows master, e.g.

Closes #6509
Closes #6833

### CODE REVIEW

**Comment by mwhooker** [comment]
I believe this is complete and ready for review

**Comment by mwhooker** [comment]
reminder to do a squash merge

File: website/source/docs/builders/oracle-classic.html.md [hunk 1]
```diff
@@ -99,6 +109,72 @@ builder. This builder currently only works with the SSH communicator.
...
+                    "ssh_username": "soandso"
+                },
+                "type": "oracle-classic"
+            }
+        ]
+    }
+    ```
+
+-   `builder_image_list` (string) - This is the image to use for the builder
+    instance. This *must* be linux instance, and Oracle Linux is recommended.
```
Comment (azr) [inline website/source/docs/builders/oracle-classic.html.md]:
```suggestion
    instance. This *must* be a linux instance, and Oracle Linux is recommended.
```

**Comment by azr** [comment]
Just found a tiny tiny typo, other than that LGTM, will test it now.

**Comment by azr** [comment]
Wohoo congrats 🎉

**Pull request merged**